src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -12,6 +12,11 @@ public static class SarifWorkItemsExtensions
     {
         public static bool ShouldBeFiled(this Result result)
         {
+            if (result == null)
```

Comment:
>result [](http://example.com/codeflow?start=16&length=6)

Why did we add this? Why wouldn't a null dereference be the expected condition?

i.e., why do we require

myObject?.ThisIsAnActualMethod()

to protect against a null variable but

myObject.ThisIsAnExtensionMethod()

should work when myObject is null?

I am asking a sincere question. But I'm not sure this change is a good design pattern, Shaopeng, what's your argument for it? :)

Reply:
I think it depends on which one should be responsible for the null check, the caller or the extension method.

If the caller should handle null value then we don't need this check inside extension method.

myObject?.ThisIsAnExtensionMethod()

Reply:
I am not sure, my thought was similar to the IDictionary cast that we were looking at last PR, either a helper already did that for you or need to cast to IDictionary everywhere it is used, 
in this case if the extension method did the check for you, 
or where you call the extension method. 
only differ when user forgot to check null before call this extension seeing an unhandled exception because we call result.BaselineState at the first line, the other case is no exception returns false.

A quick find reference I see about 10 usages of this method.

In .net source code they also do null check at the first line of the 
extension methods, like below, our logic doe not need to throw:

public static partial class StringNormalizationExtensions
    {
        public static bool IsNormalized(this string strInput, NormalizationForm normalizationForm)
        {
            if (strInput == null)
            {
                throw new ArgumentNullException(nameof(strInput));
            }

            return strInput.IsNormalized(normalizationForm);
        }

Reply:
Like the way how .net handles in the extension method. The extension method will have similar behavior to instance method and can be protected by ? operation:

myObject?.ThisIsAnExtensionMethod()

Reply:
Here's the pattern that looks weird:

((Result)null).ShouldBeFiled();